### PR TITLE
Add maximum radial distance feature

### PR DESCRIPTION
**Pull request by adrien-berchet**
Add a neurite feature to get the maximum radial distance from the soma of a collection of neurites.

### CODE REVIEW

**Comment by arnaudon** [comment]
@adrien-berchet , in the end, we don't need that, right? I guess we can close it?

**Comment by adrien-berchet** [comment]
Indeed we don't use it for now. But I am wondering if it is better to close or to merge (after rebase)? Someone might need it someday.

**Comment by arnaudon** [comment]
ah yes, we could merge, if @wizmer has nothing against it!

**Comment by wizmer** [comment]
We can merge it but you'd need to pass it as a PR (no longer a draft) and it needs a rebase

**Comment by adrien-berchet** [comment]
@wizmer I had to update `neurom.features._get_feature_value_and_func` to make it work with your new registering system but I am not sure how you intended to deal with features with `shape=()`, can you check it plz?

**Comment by arnaudon** [comment]
So `_get_feature_value_and_func` was not working previously with `shape=()` types of features?

**Comment by adrien-berchet** [comment]
Exactly, because `neurom/features/__init__.py::91` is trying to build a list from a scalar instead of an iterable.

**Comment by arnaudon** [comment]
So is this a real bug in neurom? The test were still working?

**Comment by adrien-berchet** [comment]
The tests zere zorking because the features with `shape=()` were actually not registered because they were filtered here: `neurom/features/__init__.py::177`. But I don't know the reason of this filter.

**Comment by arnaudon** [comment]
@asanin-epfl , could you merge that? Appart from adding a new feature, it fixes a small bug, see adrien last comment. Thank you!

**Comment by arnaudon** [comment]
@adrien-berchet , can you rebase so we can merge to master?

**Comment by asanin-epfl** [comment]
I will review after the rebase.

**Comment by adrien-berchet** [comment]
> I will review after the rebase.

Done

File: neurom/features/__init__.py [hunk 1]
```diff
@@ -88,7 +88,13 @@ def _get_feature_value_and_func(feature_name, obj, **kwargs):
     """
     feat = _find_feature_func(feature_name)
 
-    return np.array(list(feat(obj, **kwargs))), feat
+    res = feat(obj, **kwargs)
```
Comment (asanin-epfl) [inline neurom/features/__init__.py:91]: Why this change? Tests still pass with the old code.

Comment (adrien-berchet) [reply neurom/features/__init__.py:91]: See my two previous comments:
https://github.com/BlueBrain/NeuroM/pull/849#issuecomment-787809109
https://github.com/BlueBrain/NeuroM/pull/849#issuecomment-787846393

And there is a TODO at ``neurom/features/__init__.py::176`` so I guess this change was more or less planned.

**Comment by adrien-berchet** [comment]
I added a more usual feature so it makes it usable and points out the issue before these changes: it is not possible to use ``get_feature()`` for functions with ``shape=()``. So it is not possible to call it to just get a scalar, we always get a list of at least 1 element (I updated the behavior in the last commit).

**Review by asanin-epfl — APPROVED** [review]

**Comment by asanin-epfl** [comment]
Ok, good to go then. Thank you!

**Comment by adrien-berchet** [comment]
Thanks!

**Comment by arnaudon** [comment]
Sorry, I'm late to the party: this is fixed here: https://github.com/BlueBrain/NeuroM/pull/884
